purchase_order_block/views/purchase_order_view.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2017 Eficent Business and IT Consulting Services S.L.
+     Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+     License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). -->
+
+<odoo>
+    <record id="purchase_order_form" model="ir.ui.view">
+        <field name="name">purchase.order.form.block.reason</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form"/>
+        <field name="arch" type="xml">
+            <field name="date_order" position="after">
+                <field name="po_block_id" attrs="{'readonly': [('id','!=', False)], 'invisible':['|', ('released','=', True), ('empty_check','=', True)]}"/>
+                <field name="released" invisible='True'/>
+                <field name="empty_check" invisible='True'/>
```

Comment:
Showing always the po_block_id will make these two fields unneded.

Reply:
Agree! Removing this both boolean fields will make code cleaner and faster.

`released` If released it will make the Block Reason M2O field invisible. 
and `empty_check` will make the field invisible if it is empty upon record save.

@rgarnau What do you think? do you agree to keep the `po_block_id` field visible all the time?!

Reply:
@darshan-serpent Oke, let's leave it visible and if it is remaining blank it should be editable while the PO is in draft. This way we give more flexibility choosing a blocking rule. Could you do this?